src/client/app/utils/calibration.ts

```diff
@@ -200,3 +324,103 @@ export function normalizeImageDimensions(dimensions: Dimensions) {
+	const p: CartesianPoint = { x: xValue, y: yValue };
+	return p;
+}
+
+/**
+ * Rotates point counterclockwise through angle.
+ * @param angleDeg Angle in degrees to rotate by. Note that positive means counterclockwise.
+ * @param point (x,y) pair to rotate.
+ * @returns Rotated (x,y) pair.
+ */
+export function rotate(angleDeg: number, point: CartesianPoint): CartesianPoint {
+	// Convert angle to radians.
+	const angleRad = angleDeg / 180.0 * Math.PI;
+	// Rotate point counterclockwise. This is really a 2x2 matrix times the point vector.
+	const xRotated = point.x * Math.cos(angleRad) - point.y * Math.sin(angleRad);
+	const yRotated = point.x * Math.sin(angleRad) + point.y * Math.cos(angleRad);
+	// Rotated point to return.
+	const p: CartesianPoint = { x: xRotated, y: yRotated };
+	return p;
+}
+
+/**
+ * This shifts the point and then rotates by the angle. Typically used to go from user map to true north.
+ * @param size The size of the image (normally normalized map size)
+ * @param point The (x,y) pair for the point to be shifted & rotated
+ * @param direction The factor to scale the shift by. Normally either +1 to add the shift or -1 to subtract.
+ * @param angleDeg Angle in degrees to rotate by. Note that positive means counterclockwise.
+ * @returns (x,y) pair shifted by 1/2 the width and height of map image and rotated by angleDeg
+ */
+export function shiftRotate(size: Dimensions, point: CartesianPoint, direction: number, angleDeg: number): CartesianPoint {
+	// Note you typically shift because the rotation must occur around the center of the map but the plotly coordinates
+	// have the origin at the bottom, left corner.
+	const shifted: CartesianPoint = shift(size, point, direction);
+	// Now rotate the centered image by the angle given.
+	const rotated: CartesianPoint = rotate(angleDeg, shifted);
+	return rotated;
+}
+
+/**
+ * This rotates the point by the angle and then shifts. Typically used to go from true north to user map.
+ * @param size The size of the image (normally normalized map size)
+ * @param point The (x,y) pair for the point to be rotated & shifted
+ * @param direction The factor to scale the shift by. Normally either +1 to add the shift or -1 to subtract.
+ * @param angleDeg Angle in degrees to rotate by. Note that positive means counterclockwise.
+ * @returns (x,y) pair shifted by 1/2 the width and height of map image
+ */
+export function rotateShift(size: Dimensions, point: CartesianPoint, direction: number, angleDeg: number): CartesianPoint {
+	// Now rotate the centered image by the angle given.
+	const rotated: CartesianPoint = rotate(angleDeg, point);
+	// Note you typically shift because the rotation must occur around the center of the map but the plotly coordinates
+	// have the origin at the bottom, left corner. This shift is normally the opposite of the shiftRotate one since
+	// going the other way.
+	const shifted: CartesianPoint = shift(size, rotated, direction);
+	return shifted;
+}
+
+/**
```

Comment:
For consistency, you might want to include a one line summary of the purpose of this function.